e2e/playwright/special-permissions-actions-available/src/tests/folders-actions.spec.ts

```diff
@@ -0,0 +1,134 @@
+ */
+
+import { ApiClientFactory, FavoritesPageApi, PersonalFilesPage, NodesApi, TrashcanApi, Utils, test } from '@alfresco/playwright-shared';
+import * as testData from '@alfresco/playwright-shared';
+
+test.describe('Folders - available actions : ', () => {
+  const apiClientFactory = new ApiClientFactory();
+  const username = `user-${Utils.random()}`;
+  const parentName = `parent-${Utils.random()}`;
+
+  let parentId: string;
+  let folderFavId: string;
+  let fileFavId: string;
+
+  let nodesApi: NodesApi;
+  let favoritesActions: FavoritesPageApi;
+  let trashCanActions: TrashcanApi;
+
+  async function checkActionsAvailable(
+    myPersonalFiles: PersonalFilesPage,
+    item: string,
+    expectedToolbarPrimary: string[],
+    expectedToolbarMore: string[]
+  ): Promise<void> {
+    await myPersonalFiles.dataTable.selectItem(item);
+    await myPersonalFiles.acaHeader.verifyToolbarPrimaryActions(expectedToolbarPrimary);
+    await myPersonalFiles.acaHeader.clickMoreActions();
+    await myPersonalFiles.matMenu.verifyActualMoreActions(expectedToolbarMore);
+  }
+
+  async function checkMultipleSelActionsAvailable(
+    myPersonalFiles: PersonalFilesPage,
+    expectedToolbarPrimary: string[],
+    expectedToolbarMore: string[]
+  ): Promise<void> {
+    await myPersonalFiles.acaHeader.verifyToolbarPrimaryActions(expectedToolbarPrimary);
+    await myPersonalFiles.acaHeader.clickMoreActions();
+    await myPersonalFiles.matMenu.verifyActualMoreActions(expectedToolbarMore);
+  }
+
+  test.beforeAll(async () => {
+    test.setTimeout(140000);
+    await apiClientFactory.setUpAcaBackend('admin');
+    await apiClientFactory.createUser({ username: username });
+    nodesApi = await NodesApi.initialize(username, username);
+    favoritesActions = await FavoritesPageApi.initialize(username, username);
+    trashCanActions = await TrashcanApi.initialize(username, username);
+    parentId = (await nodesApi.createFolder(parentName)).entry.id;
+    await nodesApi.createFile(testData.file.name, parentId);
+    fileFavId = (await nodesApi.createFile(testData.fileFavFile.name, parentId)).entry.id;
+
+    await nodesApi.createFolder(testData.folderFile.name, parentId);
+    folderFavId = (await nodesApi.createFolder(testData.folderFavFile.name, parentId)).entry.id;
+
+    const initialFavoritesTotalItems = (await favoritesActions.getFavoritesTotalItems(username)) || 0;
+    await favoritesActions.addFavoritesByIds('folder', [folderFavId]);
+    await favoritesActions.addFavoritesByIds('file', [fileFavId]);
+    await favoritesActions.waitForApi(username, { expect: initialFavoritesTotalItems + 2 });
+  });
+
+  test.afterAll(async () => {
+    await nodesApi.deleteNodes([parentId]);
+    await trashCanActions.emptyTrashcan();
+  });
+
+  test.describe('on Personal Files : ', () => {
+    test('Folder not favorite  - [C213123]', async ({ personalFiles, loginPage }) => {
+      await personalFiles.navigate({ remoteUrl: `#/personal-files/${parentId}` });
```

Comment:
For every test these two lines are repeated, can we move it somehow to beforeEach or not?
await personalFiles.navigate({ remoteUrl: `#/personal-files/${parentId}` });
await loginPage.loginUser({ username: username, password: username });

Reply:
I moved this lines to beforeEach